wgengine/magicsock/peermtu_darwin.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build darwin && !ios
+
+package magicsock
+
+import (
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func getDontFragOpt(network string) int {
+	if network == "udp4" {
+		return unix.IP_DONTFRAG
+	}
+	return unix.IPV6_DONTFRAG
+}
+
+func (c *Conn) setDontFragment(network string, enable bool) error {
+	optArg := 1
+	if enable == false {
+		optArg = 0
+	}
+	rc, err := c.getSyscallConn(network)
+	if err != nil {
+		return err
+	}
+
+	rc.Control(func(fd uintptr) {
```

Comment:
Control itself might error, in which case the inner doesn't run, and we'd report nil implying success, though we were not successful

Reply:
Ooh nice catch, will fix.